asciimatics/widgets/frame.py

```diff
@@ -15,6 +15,85 @@
 from asciimatics.widgets.utilities import THEMES, logger
 
 
+class _BorderManager:
+    def __init__(self, frame, has_border, can_scroll):
+        self._frame = frame
+        self.has_border = has_border
+        self._scroll_bar = None
+        if can_scroll:
+            scroll_y = 2
+            scroll_height = frame._canvas.height - 4
+            if not has_border:
+                scroll_height = frame._canvas.height - 2
+                scroll_y = 1
+
+            self._scroll_bar = _ScrollBar(frame._canvas, frame.palette, 
+                frame._canvas.width - 1, scroll_y, scroll_height,
+                frame._get_pos, frame._set_pos, absolute=True) 
```

Comment:
The `_get_pos` and `_set_pos` funtions will also fall foul of the prospector checks.  How about we expose them asparameters to this class rather than discovered in this way?

Reply:
Now has a property called "scroll_position" with a setter/getter.

Reply:
Sadly, that doesn't address the other places that use this class.  CI caught them...

Reply:
Yeah, just saw that. Am undoing what I did and going to make the methods public.